src/mon/OSDMonitor.cc

```diff
@@ -1052,6 +1056,370 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     ::encode(pending_creatings, creatings_bl);
     t->put(OSD_PG_CREATING_PREFIX, "creating", creatings_bl);
   }
+
+  // health
+  _check_health(tmp, t);
+}
+
+void OSDMonitor::_check_health(
+  const OSDMap& nextmap,
```

Comment:
All the checks in this function are meant to be using nextmap, right?  They're looking at the old map currently.